File: node_backend/modals/home/topRestaurants.js
import mongoose from "mongoose";

const topRestaurantSchema = new mongoose.Schema({
  name: String,
  cloudinaryImageId: String,
  locality: String,
  areaName: String,
  costForTwo: String,
  cuisines: [String], // Array of strings
  avgRating: Number,
  avgRatingString: String,
  totalRatingsString: String,
  badges: Boolean,
  aggregatedDiscountInfoV3: {
    header: String,
    subHeader: String,
  },
});

const TopRestaurantModel = mongoose.model(
  "TopRestaurants",
  topRestaurantSchema
);

export default TopRestaurantModel;

File: python_backend/app/models/home/top_restaurants.py
from pydantic import BaseModel
from typing import List, Optional

class AggregatedDiscountInfoV3(BaseModel):
    header: str
    subHeader: str

class TopRestaurant(BaseModel):
    name: str
    cloudinaryImageId: str
    locality: str
    areaName: str
    costForTwo: str
    cuisines: List[str]
    avgRating: float
    avgRatingString: str
    totalRatingsString: str
    badges: bool
    aggregatedDiscountInfoV3: AggregatedDiscountInfoV3
